clang/lib/Headers/aiev2_core.h

```diff
@@ -1566,4 +1566,283 @@ inline v16accfloat addmsc_4x8_8x4( v32float v1, v32float v2, v16accfloat acc1, v
 
 #endif
 
+enum class aie_dm_resource {
+  none,
+  PM,
+  DM,
+  DM_test,
+  stack,
+  a,
+  b,
+  c,
+  d,
+  ab,
+  ac,
+  ad,
+  bc,
+  bd,
+  cd,
+  TM
+};
+
+#define AIE_DM_RESOURCE(X)                                                     \
+  __attribute__((address_space(static_cast<signed>(aie_dm_resource::X))))
+#define __aie_dm_resource_stack AIE_DM_RESOURCE(stack)
+#define __aie_dm_resource_a AIE_DM_RESOURCE(a)
+#define __aie_dm_resource_b AIE_DM_RESOURCE(b)
+#define __aie_dm_resource_c AIE_DM_RESOURCE(c)
+#define __aie_dm_resource_d AIE_DM_RESOURCE(d)
+#define __aie_dm_resource_ab AIE_DM_RESOURCE(ab)
+#define __aie_dm_resource_ac AIE_DM_RESOURCE(ac)
+#define __aie_dm_resource_ad AIE_DM_RESOURCE(ad)
+#define __aie_dm_resource_bc AIE_DM_RESOURCE(bc)
+#define __aie_dm_resource_bd AIE_DM_RESOURCE(bd)
+#define __aie_dm_resource_cd AIE_DM_RESOURCE(cd)
+#define __aie_dm_resource_TM AIE_DM_RESOURCE(TM)
```

Comment:
Could you add test for these to see how they get translated into IR? Not sure we need everything, but at least a. b, c and d would be nice.

Reply:
Sure, have added to IR translation test for all the dm_resources.